terraform/setup/main.tf

```diff
@@ -60,4 +60,26 @@ resource "aws_dynamodb_table" "performance-dynamodb-table" {
     read_capacity   = 10
     projection_type = "ALL"
   }
-}
\ No newline at end of file
+}
+
+## Setup Dedicated Host for Mac Resources
+## https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_host
+## It is a requirement before creating an EC2 Mac Host
+## https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-mac-instances.html
+## Moreover, you can only place 1 mac instance on a dedicate host a single time.
+## Therefore, limit down the scope for testing in Mac since EC2 can be done with Linux
+## and Mac under the hood share similar plugins with Linux
+## Mac has scrubbing to avoid patching 
+## https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-mac-instances.html#mac-instance-stop
+resource "aws_ec2_host" "dedicated_host" {
```

Comment:
Does this run every time we execute the mac test? This may be a problem since we can only tear down 24 hr in the future.

Reply:
Nope. It's run once since its a setup script.

Reply:
Changing it to 4 for avoid the unknown unknown